conf/spgwu.bess

```diff
@@ -172,9 +190,30 @@ class Port:
 
                 # Direct fast path traffic to Frag module
                 merge = __bess_module__("{}Merge".format(name), 'Merge')
-                frag = __bess_module__("{}IP4Frag".format(name), 'IPFrag')
-                merge -> frag:1 -> self.fpo
-                frag:0 -> Sink()
+
+                out = self.fpo
+
+                # Attach frag module (if enabled)
+                if ip_frag_with_eth_mtu is not None:
+                    frag = __bess_module__("{}IP4Frag".format(name), 'IPFrag', mtu=ip_frag_with_eth_mtu)
+                    frag:1 -> out
+                    frag:0 -> Sink()
+                    out = frag
+
+                # Attach telemeric module (if enabled)
+                if measure:
+                    m = Measure(latency_ns_resolution=1, latency_ns_max=100000, offset=-(self.msr_off2))
+                    m -> out
+                    out = m
+
+                # Attach nat module (if enabled)
+                if self.ext_addr is not None:
+                    self.nat = __bess_module__("{}NAT".format(conf[interface]["ifname"]), 'NAT', ext_addrs=[{'ext_addr': ext_addr}])
```

Comment:
we need port range too. Maybe 10001-60000 (not user-configurable). also list of ext_addr instead of one.

Reply:
I agree that we may need more ip addresses. Not sure about port ranges.. since NAT automatically assigns port numbers within the module (the module is already optimized to do so).

Reply:
Multiple NAT IP addresses are now supported.